client/me/concierge/calendar-card.js

```diff
@@ -68,6 +82,41 @@ class CalendarCard extends Component {
 		);
 	};
 
+	onChange = ( { target } ) => {
+		this.props.selectConciergeTimeSlot( this.props.date, target.value );
+	};
+
+	submitForm = () => {
+		const { signupForm } = this.props;
+		const timeSlots = this.props.selectedTimeSlots;
+		const meta = {
+			message: signupForm.message,
+			timezone: signupForm.timezone,
+		};
+
+		if ( timeSlots[ this.props.date ] ) {
+			this.props.bookConciergeAppointment(
+				WPCOM_CONCIERGE_SCHEDULE_ID,
+				timeSlots[ this.props.date ],
+				this.props.currentUserId,
+				this.props.site.ID,
+				meta
+			);
+		}
+	};
+
+	componentWillMount() {
+		// pre-select first dropdown option
+		this.props.selectConciergeTimeSlot( this.props.date, this.props.times[ 0 ] );
+	}
+
+	componentWillUpdate( nextProps ) {
+		if ( nextProps.bookingStatus === CONCIERGE_STATUS_BOOKED ) {
```

Comment:
Because this status change is not scoped to any particular day, every mounted `CalendarCard` will go through `componentWillUpdate` and this will be `true` for all of them. That means this condition can theoretically run 7 times. 

In this particular case you don't see this effect because the `this.props.onSubmit()` below ultimately calls a function in `ConciergeMain` which moves on to the next page, unmounting all these Cards. So only one of them hits the `componentWillUpdate` before the rest are unmounted. But if we change anything about this setup, it's likely that we'll get 7 of the same success handlers firing.

Instead, `CalendarCard` should continue to be a dumb component, and it should simply pass the timestamp selected here up to the `ConciergeStep` in the `onSubmit` prop. The `ConciergeStep` should then be responsible for taking that timestamp and the rest of the data from Redux and dispatching the booking action.

Reply:
I also want to note — I'm not sure that you even need to keep the selected timestamps in Redux. I know right now it's all keyed on the specific day so that each `CalendarCard` can remember which time was selected on that day. But that Redux data is not feeding back into the UI at all, and the only advantage to keeping it around is to maintain the selected time on refresh — but that's a very quick thing to select so I'm not very concerned with state persistence on that.

Instead, you could add component `state` to the `CalendarCard` to track the currently-selected timestamp for that card. And then when the customer clicks the "Book this session" button, you call `this.props.onSubmit( this.state.timestamp )` to send it back up to the `CalendarStep`. The `CalendarStep` then has all the data it needs to request the booking.